examples/cart/index.js

```diff
@@ -163,62 +202,111 @@ $(function() {
   /* Close Cart
   ============================================================ */
   function closeCart() {
-    $('.cart .btn--close').click(function () {
-      $('.cart').removeClass('js-active');
+    $('.cart').removeClass('js-active');
+    $('.overlay').removeClass('js-active');
+  }
+
+  /* Determine action for variant adding/updating/removing
+  ============================================================ */
+  function variantToCartHandler(variant, quantity) {
+    openCart();
+    var foundInCart = cart.lineItems.filter(function (item) {
+      return (item.variant_id === variant.id);
     });
+
+    if (foundInCart.length > 0) {
+      updateVariantInCart(variant, quantity, foundInCart[0]);
+    } else {
+      addVariantToCart(variant, quantity);
+    }
+
+    updateCartTabButton();
   }
 
+  /* Update details for item already in cart. Remove if necessary
+  ============================================================ */
+  function updateVariantInCart(variant, quantity, cartLineItem) {
+    cart.addVariants({ variant: variant, quantity: quantity }).then(function() {
+      var $cartItem = $('.cart').find('.cart-item[data-variant-id="' + variant.id + '"]');
+
+      if (cartLineItem.quantity > 0) {
+        $cartItem.find('.cart-item__quantity').attr('value', cartLineItem.quantity);
+        $cartItem.find('.cart-item__price').text(formatAsMoney(cartLineItem.line_price));
+      } else {
+        $cartItem.addClass('js-hidden').bind('transitionend webkitTransitionEnd oTransitionEnd MSTransitionEnd', function() {
+           $cartItem.remove();
+        });
+      }
+
+      updateTotalCartPricing();
+
+    }).catch(function (errors) {
+      console.log('Fail');
+      console.error(errors);
+    });
+
+  }
 
   /* Add 'quantity' amount of product 'variant' to cart
   ============================================================ */
   function addVariantToCart(variant, quantity) {
     openCart();
 
     cart.addVariants({ variant: variant, quantity: quantity }).then(function() {
-      renderCartItems();
+      var $cartItem = renderCartItem(cart.lineItems[cart.lineItems.length - 1]);
```

Comment:
`addVariantToCart` adds a variant that currently doesn't exist in the cart, and is therefore the last item in the list. So we can just grab the last item and render that instead of doing a search for it first.